front/lib/models/assistant/actions/process.ts

```diff
@@ -0,0 +1,176 @@
+import type { ProcessSchemaPropertyType, TimeframeUnit } from "@dust-tt/types";
+import type {
+  CreationOptional,
+  ForeignKey,
+  InferAttributes,
+  InferCreationAttributes,
+} from "sequelize";
+import { DataTypes, Model } from "sequelize";
+
+import { AgentConfiguration } from "@app/lib/models/assistant/agent";
+import { frontSequelize } from "@app/lib/resources/storage";
+
+export class AgentProcessConfiguration extends Model<
+  InferAttributes<AgentProcessConfiguration>,
+  InferCreationAttributes<AgentProcessConfiguration>
+> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+
+  declare agentConfigurationId: ForeignKey<AgentConfiguration["id"]>;
+
+  declare sId: string;
+
+  declare relativeTimeFrame: "auto" | "none" | "custom";
+  declare relativeTimeFrameDuration: number | null;
+  declare relativeTimeFrameUnit: TimeframeUnit | null;
+
+  declare schema: ProcessSchemaPropertyType[];
+
+  declare forceUseAtIteration: number | null;
+}
+
+AgentProcessConfiguration.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    createdAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    updatedAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    sId: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    relativeTimeFrame: {
+      type: DataTypes.STRING,
+      allowNull: false,
+      defaultValue: "auto",
+    },
+    relativeTimeFrameDuration: {
+      type: DataTypes.INTEGER,
+      allowNull: true,
+    },
+    relativeTimeFrameUnit: {
+      type: DataTypes.STRING,
+      allowNull: true,
+    },
+    schema: {
+      type: DataTypes.JSONB,
+      allowNull: false,
+    },
+    forceUseAtIteration: {
+      type: DataTypes.INTEGER,
+      allowNull: true,
+    },
+  },
+  {
+    modelName: "agent_process_configuration",
+    sequelize: frontSequelize,
+    hooks: {
+      beforeValidate: (p: AgentProcessConfiguration) => {
+        // Validation for Timeframe
+        if (p.relativeTimeFrame == "custom") {
```

Comment:
```suggestion
        if (p.relativeTimeFrame === "custom") {
```

Reply:
:+1: fixed here and where it was cargo culted from